### PR TITLE
Flash manager

### PR DESCRIPTION
**Pull request by jbleyel**

### CODE REVIEW

File: lib/python/Screens/FlashManager.py [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+						}
+				except Exception:
+					print("[FlashManager] Error: Unable to extract file list from Zip file '%s'!" % file)
+
+		if not self.imagesList:
+			try:
+				feedURL, boxInfoField = FEED_URLS.get(self.imageFeed, ("http://images.mynonpublic.com/openatv/json/%s", "machinebuild"))
+				self.box = BoxInfo.getItem(boxInfoField, "")
+				url = feedURL % self.box
+				req = Request(url, None, {'User-agent' : 'Mozilla/5.0 (Windows; U; Windows NT 5.1; de; rv:1.9.1.5) Gecko/20091102 Firefox/3.5.5'})
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:124]: Is this URL requesting a German language version of the page?  It should be requesting US English or English at least.

I would also use double quotation marks and correct the spacing before the colon:
req = Request(url, None, {"User-agent": "Mozilla/5.0 (Windows; U; Windows NT 5.1; de; rv:1.9.1.5) Gecko/20091102 Firefox/3.5.5"})

File: lib/python/Screens/FlashManager.py [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+
+	def keyDistribution(self):
+		default = 0
+		distributionList = []
+		for index, distribution in enumerate(sorted(FEED_URLS.keys())):
+			distributionList.append((distribution, distribution))
+			if distribution == self.imageFeed:
+				default = index
+		message = _("Please select a distribution from which you would like to flash an image:")
+		self.session.openWithCallback(self.keyDistributionCallback, MessageBox, message, default=default, list=distributionList, windowTitle=_("Flash Manager"), simple=True)
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:198]: Why is the "simple" form being requested.  I thought we were going to remove this.

File: lib/python/Screens/FlashManager.py [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+			"cancel": (self.abort, _("Add help text")),
+			"ok": (self.ok, _("Add help text"))
+		}, prio=-1)
+		self["progress"] = ProgressBar()
+		self["progress"].setRange((0, 100))
+		self["progress"].setValue(0)
+		self.delay = eTimer()
+		self.delay.callback.append(self.confirmation)
+		self.delay.start(0, True)
+		self.hide()
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:290]: Why has
	self.callLater(self.confirmation)
been removed and been replaced with functionally equivalent code?
	self.delay = eTimer()
	self.delay.callback.append(self.confirmation)
	self.delay.start(0, True)

This necessitates the inclusion of eTimer() into this code when it is not really required.

File: lib/python/Screens/FlashManager.py [hunk 4]
```diff
@@ -0,0 +1,600 @@
...
+			choices = ((_("Upgrade (Backup, Flash & Restore All)"), "restoresettingsandallplugins"),
+			(_("Clean (Just flash and start clean)"), "wizard"),
+			(_("Backup, flash and restore settings and no plugins"), "restoresettingsnoplugin"),
+			(_("Backup, flash and restore settings and selected plugins (ask user)"), "restoresettings"),
+			(_("Do not flash image"), "abort"))
+			self.session.openWithCallback(self.postFlashActionCallback, ChoiceBox, title=title, list=choices, selection=self.SelectPrevPostFlashAction())
+		else:
+			self.abort()
+
+	def SelectPrevPostFlashAction(self):
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:403]: Method names should start with a lower case character (camelCase).

File: lib/python/Screens/FlashManager.py [hunk 5]
```diff
@@ -0,0 +1,600 @@
...
+		self.delay.start(0, True)
+
+	def doUnzip(self):
+		try:
+			ZipFile(self.zippedimage, "r").extractall(self.unzippedimage)
+			self.doFlashImage()
+		except:
+			self.session.openWithCallback(self.abort, MessageBox, _("Error during unzipping image\n%s") % self.imagename, type=MessageBox.TYPE_ERROR, simple=True)
+
+	def doFlashImage(self):
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:528]: What was wrong with "flashImage(self):"?

File: lib/python/Screens/FlashManager.py [hunk 6]
```diff
@@ -0,0 +1,600 @@
...
+				self.MTDKERNEL = BoxInfo.getItem("mtdkernel")
+				self.MTDROOTFS = BoxInfo.getItem("mtdrootfs")
+			CMD = [OFGWRITE, OFGWRITE]
+			if BoxInfo.getItem("model") in ("dm820", "dm7080"): # temp solution ofgwrite autodetection not ready
+				CMD.append("-rmmcblk0p1")
+			elif self.MTDKERNEL == self.MTDROOTFS:	# receiver with kernel and rootfs on one partition
+				CMD.append("-r")
+			else: # normal non multiboot receiver
+				CMD.append("-r")
+				CMD.append("-k")
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:558]: This could be an "extend()".

File: lib/python/Screens/FlashManager.py [hunk 7]
```diff
@@ -0,0 +1,600 @@
...
+			if MultiBoot.canMultiBoot():
+				if (self.ROOTFSSUBDIR) is None:	# receiver with SD card multiboot
+					CMD.append("-r%s" % self.MTDROOTFS)
+					CMD.append("-k%s" % self.MTDKERNEL)
+					CMD.append("-m0")
+				else:
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:564]: This could be an "extend()".

File: lib/python/Screens/FlashManager.py [hunk 8]
```diff
@@ -0,0 +1,600 @@
...
+					CMD.append("-r")
+					CMD.append("-k")
+					CMD.append("-m%s" % self.multibootslot)
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:567]: This could be an "extend()".

File: lib/python/Screens/FlashManager.py [hunk 9]
```diff
@@ -0,0 +1,600 @@
...
+			elif self.MTDKERNEL == self.MTDROOTFS:	# receiver with kernel and rootfs on one partition
+				CMD.append("-r")
+			else: # normal non multiboot receiver
+				CMD.append("-r")
+				CMD.append("-k")
+			if MultiBoot.canMultiBoot():
+				if (self.ROOTFSSUBDIR) is None:	# receiver with SD card multiboot
+					CMD.append("-r%s" % self.MTDROOTFS)
+					CMD.append("-k%s" % self.MTDKERNEL)
+					CMD.append("-m0")
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:563]: This could be an "extend()".

File: lib/python/Screens/FlashManager.py [hunk 10]
```diff
@@ -0,0 +1,600 @@
...
+				else:
+					CMD.append("-r")
+					CMD.append("-k")
+					CMD.append("-m%s" % self.multibootslot)
+			CMD.append("'%s'" % imagefiles)
+			self.containerofgwrite = Console()
+			self.containerofgwrite.ePopen(CMD, self.flashImageDone)
+			fbClass.getInstance().lock()
+		else:
+			self.session.openWithCallback(self.abort, MessageBox, _("Image to install is invalid\n%s") % self.imagename, type=MessageBox.TYPE_ERROR, simple=True)
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:573]: Aren't we getting rid of "simple" message boxes?

**Review by IanSav — APPROVED** [review]
Please check for proper PEP8 compliance:
1) Comments have 2 spaces from the code to the "#" and then one space after.  As a style I suggest the comments should be in proper English US with a captial letter first and a full stop at the end.
2) Please use PEP8 line spacing throughout the code.
3) Please use correct indentation for multi-line commands.
4) There should be NO naked "exception" statements.  I posted an example correction on line 441.

File: lib/python/Screens/FlashManager.py [hunk 11]
```diff
@@ -0,0 +1,600 @@
...
+				if bootSlots[self.multibootslot].get("ubi"):
+					self.MTDROOTFS = bootSlots[self.multibootslot]["device"]
+				else:
+					self.MTDROOTFS = bootSlots[self.multibootslot]["device"].split("/")[2]
+				if MultiBoot.hasRootSubdir():
+					self.ROOTFSSUBDIR = bootSlots[self.multibootslot]["rootsubdir"]
+			else:
+				self.MTDKERNEL = BoxInfo.getItem("mtdkernel")
+				self.MTDROOTFS = BoxInfo.getItem("mtdrootfs")
+			CMD = [OFGWRITE, OFGWRITE]
```
Comment (IanSav) [inline lib/python/Screens/FlashManager.py:551]: CMD is not a global variable so normal camelCase should be used.  In this case the name might be "cmd".

File: lib/python/Screens/FlashManager.py (hunk 3 above)
Comment (jbleyel) [reply lib/python/Screens/FlashManager.py:290]: Check line 517

File: lib/python/Screens/FlashManager.py (hunk 5 above)
Comment (jbleyel) [reply lib/python/Screens/FlashManager.py:528]: all other functions are also doXXX

Comment (jbleyel) [reply lib/python/Screens/FlashManager.py:528]: all other functions are also doXXX

**Pull request merged**